Lib/glyphsLib/builder/sources.py

```diff
@@ -80,6 +83,59 @@ def _to_designspace_source(self, master, is_regular):
     source.location = location
 
 
+def _to_designspace_source_layer(self):
+    brace_layers = collections.defaultdict(list)
+    layer_to_glyph = collections.defaultdict(list)
+    for glyph in self.font.glyphs:
+        for layer in glyph.layers:
+            if (
+                "{" in layer.name
+                and "}" in layer.name
+                and ".background" not in layer.name
+            ):
+                associated_master = self._sources[layer.associatedMasterId]
+                brace_layers[layer].append(associated_master)
+                layer_to_glyph[layer.name].append(glyph.name)
+
+    designspace = self._designspace
+    for brace_layer, masters in brace_layers.items():
+        brace_layer_name = brace_layer.name
+        layer_coordinates = [
+            int(c)
+            for c in brace_layer_name[
+                brace_layer_name.index("{") + 1 : brace_layer_name.index("}")
+            ].split(",")
+        ]
+        if len(layer_coordinates) != len(designspace.axes):
+            logger.warning(
+                "Glyph(s) %s, brace layer '%s' defines more or less locations than "
+                "there are design axes.",
+                layer_to_glyph[brace_layer_name],
+                brace_layer_name,
+            )
+
+        layer_coordinates_mapping = {}
+        for axis, location in zip(designspace.axes, layer_coordinates):
+            if not (axis.minimum <= location <= axis.maximum):
+                raise ValueError(
+                    "Glyph(s) {}, brace layer '{}', location {} is out of bounds for "
+                    "axis {}".format(
+                        layer_to_glyph[brace_layer_name],
+                        brace_layer_name,
+                        location,
+                        axis.name,
+                    )
+                )
+            layer_coordinates_mapping[axis.name] = location
+
+        for master in masters:
+            s = fontTools.designspaceLib.SourceDescriptor()
+            s.filename = master.filename
+            s.layerName = brace_layer_name
+            s.location = layer_coordinates_mapping
+            designspace.sources.append(s)
```

Comment:
here we need to make sure that we don't add duplicated sources (with same filename, same layer name, and same location values)